gui/src/lib/api.ts

```diff
@@ -289,6 +292,18 @@ export const api = (() => {
         await invoke('refresh-all-components');
       },
 
+      async getComponentDetails(): Promise<ComponentDetails[]> {
+        const processes = await this.describeProcesses();
+        return await Promise.all(
```

Comment:
please make `get-component-status` a bulk operation instead of doing `Promise.all` here.